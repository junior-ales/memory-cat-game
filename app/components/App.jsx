import React from 'react';
import { curry, reduce } from 'ramda';

import Header from './Header';
import TodoList from './TodoList';
import Footer from './Footer';

const combineComponents = (...args) => {
  const combine = curry((c, o) => x => (
    <section>{c(x)} {o(x)}</section>
  ));

  const [first, ...rest] = args;
  return reduce((acc, c) => combine(acc, c), first, rest);
};

export default combineComponents(Header, TodoList, Footer);
