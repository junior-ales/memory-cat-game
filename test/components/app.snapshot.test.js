/* eslint-disable */
import renderer from 'react-test-renderer';
import React from 'react';
import App from '../../app/components/App';

const props = {
  year: '2016',
  title: 'Reat & Redux Todo List',
  todos: [
    { id: 1, text: 'foo', completed: false },
    { id: 2, text: 'bar', completed: false }
  ],
  dispatch: fn => x => fn(x)
};

test('should render the application', () => {
  const app = renderer.create(<App {...props} />);
  expect(app.toJSON()).toMatchSnapshot();
});
