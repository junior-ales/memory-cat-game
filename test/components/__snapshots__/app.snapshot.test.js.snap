exports[`test should render the application 1`] = `
<section>
  <section>
    <header>
      <h1>
        A Todo List: 
        Reat & Redux Todo List
      </h1>
      <button
        onClick={[Function]}>
        Add
      </button>
    </header>
     
    <ul>
      <li>
        foo
        <button
          onClick={[Function]}>
          Remove
        </button>
      </li>
      <li>
        bar
        <button
          onClick={[Function]}>
          Remove
        </button>
      </li>
    </ul>
  </section>
   
  <div>
    2016
  </div>
</section>
`;
